### ISSUE TITLE
[Usability] Making My Websites / Installed Applications clearer

### ISSUE THREAD

**Issue by Bugsbane** [cznweb/genesis#453]
**THE ISSUES:**
At the moment, there are a few things that get a bit confusing with the arkOS interface regarding apps vs installs of apps vs websites. Notably:
- If someone clicks on "My applications" they don't actually see the applications they've set up, just the plugins they've installed
- If someone has say, installed the WordPress, Ghost and ownCloud plugins and set up, say, WordPress and ownCloud. Consider what happens when they click on My Applications > **Ghost**. They're taken to a page that doesn't show Ghost at all. It just shows WordPress and ownCloud. Huh?
- When making alterations to websites, there is no direct link to the websites, making people repeatedly have to find the "My Websites" page without any link or anything saying that if they click on it they'll go to "My Websites". Even when they know where it is, it requires clicking through "My Applications" first _every single time_, even when they have no desire to change anything on the "My Applications" page. This is one of the most frequently used pages in arkOS, but there is no quick and easy way to find it and then access it.
- While "My Websites" currently doesn't have _any_ direct link, Installed plugins are currently shown and listed in _two_ places, _both_ with a direct, clearly labled link, namely "My Applications" and "App Store" > Installed.
- The difference between applications / plugins and websites isn't always clear in the current UI. Currently (and still in my mockup below, unfortunately) we talk of "Installed applications" and on the same page have a button referring to the same list to "Update plugin list". So what's the difference between a plugin and an application here? If an application is an installed instance of a plugin, then what's the difference between an owncloud application and an owncloud website?

**SUGGESTIONS:**
The "My Applications" page is only used to direct people to "My Webpages" or to choose a plugin to configure. We can scrap the "My Applications" page altogether if we:
1. Replace the "My Applications" menu link with a "My Websites" link.
2. Add a "configure" icon in the "App Store" > Installed page. The name of the app could also go to the configure page (the same as clicking the app in "My Applications" currently).
3. Make the App Store > Installed actions more consistent with the other page layouts such as the Networking / Hosts / DNS / Firewall Apps / Defense pages, by using the icon on the left, and all of the actions as icons to the right. In this case the three action icons would be the enabled/disabled tick, the configure icon (spanner?), the delete icon (I would suggest the trash bin instead of the circle with the X, but whatever) and an "update" icon where upgrades are available (not shown in mockup). If multiple updates are available, add an "Upgrade All" button next to "Update Plugin List" (also not shown in mockup).
4. Rename "App Store" to "Applications" to reflect that it includes currently installed apps, and make the "Installed" tab the first, default one. Add a clear "Add new App" button, that just goes to the "Available" tab to aid discoverability of the available apps. On a side note, are you aware that Apple went and forced a similar project (Amahi) to change the name of their App Store which does exactly the same thing as arkOS'?
5. If a plugin is installed but has no website yet, it's configure icon is greyed out, with a hover tooltip that says "Add a website using this plugin to configure it." If there is an instance installed, say of ownCloud, then go to the "My Wesites" page, listing the ownCloud installs first. Possibly other apps could be greyed out but still clickable.
6. Consistently call plugins / applications by one name. I suggest we drop using the phrase "plugin" and just call them applications, especially given that many applications (eg WordPress) have their own plugins. Talking about plugins' plusgins will just get confusing. Many people know the phrase web applications or web apps though.

So in short:
Application == plugin
Website == installed, user facing application.
Plugin == not mentioned in the interface

**MOCKUP** (Applications page. "My Websites" goes to current "My Websites" page):
![ui-cleanup](https://cloud.githubusercontent.com/assets/401733/4005516/51a22d5a-2993-11e4-97c2-72728a852a44.png)

**Comment by peakwinter** [comment]
@Bugsbane In 0.7 many of your recommendations have been implemented or at least partially addressed. I'd be eager to hear what you think of the new flow after having had some time to test it. I will move this point to 0.8 to follow up later.

**Comment by Bugsbane** [comment]
Thanks! I'll take a look, and post back here.

**Comment by Bugsbane** [comment]
Ok, I've taken a look. I have a couple of thoughts, although generally speaking, it's difficult to give feedback yet, as currently bugs get in the way of too many processes to really see how the users interaction flow would be.

Currently though, I would suggest:

* In "My applications" that after all of the installed applications there was a box with a + sign and "Add a new application" which just links to /apps

* If someone clicks on an app, say ownCloud, instead of just going to a generic "My Websites" which shows unrelated apps like Ghost, or Mailpile, instead check if there is an install of that app already. If there isn't, bring up the page users see when they click "Add website" with that app preselected. It's the only interaction they can do anyway, why make them search for the button?

If they *do* have >0 installs of that app, list *only* the installs for that app they chose. In other words, if someone clicked "ownCloud", *only* show them the ownCloud installs (not Ghost or Mailpile). Title it "My ownCloud Websites". It's confusing to click on one app, and be given listings for some other app instead. For example, right now, if someone had only created a Ghost website previously, and then clicked on the ownCloud plugin, they would have clicked ownCloud, but be shown just a Ghost install, and see no mention of ownCloud at all. Not intuitive.

After the boxes displaying their current installs, add another with a + icon and "Add new ownCloud website" (or whatever app) just like in my previous point, to keep the UI consistent.

Below that, if you wished, you could have a link for "Show all websites" at the bottom which would show the same results we see in /websites now.

I'm assuming it's still too early to report bugs that don't prevent Genesis starting up (such as not being able to install ownCloud because the MySQL daemon didn't get started automatically).

**Comment by peakwinter** [comment]
Hmm, the commit to fix that last point actually got added in last night... Does your box date to before this release mentioned [here](https://bbs.arkos.io/t/official-0-7-beta-hype-thread/1649/51?u=peakwinter)?

Feel free to start reporting bugs you run into in 0.7 :) All help is appreciated!

**Comment by Bugsbane** [comment]
>  Does your box date to before this release mentioned here?

I downloaded the version in the second post. I guess there's a chance that I may have accidentally installed the first version again. I'll check the box MD5 and reinstall.

**Comment by peakwinter** [comment]
Ok cool! Also don't forget:

> A reminder that, if you download this and use it in the same folder you've used with a prior box, you must use vagrant destroy first, then create the box with a new name

**Comment by Bugsbane** [comment]
Thanks. Assuming the md5 is correct, I'll vagrant destroy, delete .vagrant and .vagrantfile and reinit the box and report back.

**Comment by Bugsbane** [comment]
> the commit to fix that last point actually got added in last night...

Umm... When I click on MailPile, I'm still seeing non-MailPile apps listed (specifically Ghost)

**Comment by peakwinter** [comment]
The last point meant:

> (such as not being able to install ownCloud because the MySQL daemon didn't get started automatically)

And yes you are probably right -- you have to add a domain name via the Users button, Domains tab. Then you can add that address to the site. I will make a popup to make this clear if there are no domains added.

Thanks!

**Comment by Bugsbane** [comment]
Rather than a popup making them go elsewhere in the UI, if you know they need to create a domain, why not just start the process? For example, they click to create an ownCloud website and it immediately asks them for a domain name to create, and then goes to the website configuration screen.

Even better than that (although more coding) would be an option somewhat like Google's search suggestions.

In other words, let people start typing the domain name in a text area on the website configuration screen, and as they do that, do some Ajaxy Magic™ to show any pre-created domains that the current text area string would match, eg, if the user had typed rko and created a domain arkos.org it would show as a suggestion, because the string rko can be found in a**rko**s.org Then they can just click on the domain they want, or leave it to create a new one.

**Comment by peakwinter** [comment]
Might be a good idea for a future widget. At this point the user is given the opportunity to add their first domain during the First Run wizard but that doesn't come up in the dev image :)

**Comment by Bugsbane** [comment]
Ok. That would make it better. What about setting up their second, third, fourth. etc. hosts / domains though?
